components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.spec.ts

```diff
@@ -0,0 +1,177 @@
+        ReactiveFormsModule,
+        StoreModule.forRoot(ngrxReducers, { runtimeChecks })
+      ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateEnvironmentModalComponent);
+    component = fixture.componentInstance;
+    component.firstFormGroup = new FormBuilder().group({
+      name: ['', null],
+      description: ['', null]
+    });
+    component.thirdFormGroup = new FormBuilder().group({
+      dattr: ['', null]
+    });
+    component.fourthFormGroup = new FormBuilder().group({
+      oattr: ['', null]
+    });
+    component.openEvent = new EventEmitter();
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+
+  describe('create environment', () => {
+    let store: Store<NgrxStateAtom>;
+    const environment: Environment = {
+      org_id: 'chef_manage',
+      server_id: 'test',
+      name: 'test',
+      description: 'test environment',
+      cookbook_versions: {
+        aix: '> 2.3.4'
+      },
+      default_attributes: {
+        test: 'test'
+      },
+      override_attributes: {
+        test: 'test'
+      }
+    };
+
+    beforeEach(() => {
+      store = TestBed.inject(Store);
+    });
+
+    it('opening creat model and defaults tab to showing section', () => {
+      component.visible = true;
+      expect(component.detailsTab).toBe(true);
+      expect(component.constraintsTab).toBe(false);
+      expect(component.defaultTab).toBe(false);
+      expect(component.overrideTab).toBe(false);
+    });
+
+    it('opening create modal resets to empty string', () => {
+      component.visible = true;
+      expect(component.firstFormGroup.controls['name'].value).toEqual('');
+      expect(component.firstFormGroup.controls['description'].value).toEqual('');
+      expect(component.thirdFormGroup.controls['dattr'].value).toEqual('{}');
+      expect(component.fourthFormGroup.controls['oattr'].value).toEqual('{}');
+    });
+
+    it('on conflict error, modal remains open and displays conflict error', () => {
+      spyOn(component.conflictErrorEvent, 'emit');
+      component.visible = true;
+      component.firstFormGroup.controls['name'].setValue(environment.name);
+      component.firstFormGroup.controls['description'].setValue(environment.description);
+      component.constraintArray = environment.cookbook_versions;
+      component.thirdFormGroup.controls['dattr'].setValue(
+        JSON.stringify(environment.default_attributes));
+      component.fourthFormGroup.controls['oattr'].setValue(
+        JSON.stringify(environment.override_attributes));
+
+      component.createEnvironment();
+
+      const conflict = <HttpErrorResponse>{
+        status: HttpStatus.CONFLICT,
+        ok: false
+      };
+      store.dispatch(new CreateEnvironmentFailure(conflict));
+      expect(component.visible).toBe(true);
+    });
+
+    it('on success, closes modal and adds new environment', () => {
```

Comment:
You are not confirming that the modal closes--you are actually closing the modal manually in the test here.

Reply:
Updated the testcase